pkg/collector/node.go

```diff
@@ -38,12 +57,27 @@ func (n *NodeCollector) Collectors() map[string]collector.Collector {
 
 // Name returns the name of this collector
 func (n *NodeCollector) Name() string {
-	return "do-agent"
+	return "node"
 }
 
 // Collect collects metrics using prometheus/node_exporter
 func (n *NodeCollector) Collect(ch chan<- prometheus.Metric) {
-	n.collectFunc(ch)
+	tee := make(chan prometheus.Metric, 1)
+	go func() {
+		defer close(tee)
+		n.collectFunc(tee)
+	}()
+	for m := range tee {
+		// Desc doesn't allow access to underlying fields like fqName. The String() output contains
+		// Desc{fqName: "node_network_transmit_bytes_total", help: "Network device statistic transmit_bytes.", constLabels: {}, variableLabels: [device]}
+		// this is ugly but currently all we can do
+		d := strings.ToLower(m.Desc().String())
+		for s := range whitelist {
```

Comment:
parse out the fqname and do a map lookup instead.

Alternatively you can do something like 
```
metric := &dto.Metric{}
m.Write(metric)
```
and pase the metric labels for `__name__` to determine the metric name.

Reply:
If I parse it out I'll have to use regex which is easily tenfold slower than just a string search. I'll try with creating a dto.Metric

Reply:
Update: that introduces quite a bit more code by having to check for errors on the Write, looping on label values, etc. What I have feels like the simplest solution :slightly_frowning_face:

Reply:
it might be slower, but it's probably a more "correct" way to do things.

Reply:
I mean, simple and incorrect vs. complex and correct.

Reply:
After attempts to use the m.Write and other methods I haven't found a way to access the fqName.